### DISCUSSION TITLE
Support for new specifications in OWL 2

### DISCUSSION THREAD

**Discussion by DhanshreeA** [RDFLib/rdflib#1567] [Ideas]
Does rdflib currently support adding data ranges as per OWL2 specifications? For Example, can you create a custom datatype like: 

> ```
> g.add((URIRef(ns+"Answer"), RDF.type, RDFS.Datatype))
> g.add((URIRef(ns+"Answer"), OWL.unionOf, (XSD.string, XSD.boolean, XSD.positiveInteger))) 
> g.add((URIRef(ns+"answer_value"), RDF.type, RDF.Property))
> g.add((URIRef(ns+"answer_value"), RDFS.domain, URIRef(ns+"Question")))
> g.add((URIRef(ns+"answer_value"), RDFS.range, URIRef(ns+"Answer")))
> ```

**Comment by MuraliKrishna26** [comment]
Hey DhansheeA, I have worked on the issue and fixed it. Now you can add more than one data-type(or any property) at once as you mentioned in the code.
The below line will work fine 
       g.add((URIRef(ns+"Answer"), OWL.unionOf, (XSD.string, XSD.boolean, XSD.positiveInteger))

I have observed there may be a typo in the line : 
       g.add((URIRef(ns+"answer_value"), RDF.type, RDFS.Property))
Property term is defined in RDF class but not in RDFS.

**Reply by nikokaoja** [reply]
@MuraliKrishna26 , I tried:
g.add((URIRef(ns+"Answer"), OWL.unionOf, (XSD.string, XSD.boolean, XSD.positiveInteger))

and it does not work. I am getting:

```
AssertionError: Object (rdflib.term.URIRef('http://www.w3.org/2001/XMLSchema#string'), rdflib.term.URIRef('http://www.w3.org/2001/XMLSchema#boolean'), rdflib.term.URIRef('http://www.w3.org/2001/XMLSchema#positiveInteger')) must be an rdflib term
```

**Comment by ashleysommer** [comment]
@MuraliKrishna26 
> Property term is defined in RDF class but not in RDFS.

RDF.type is correct. Even though it is specified in the RDFS spec, the `type` predicate _intentionally_ uses `rdf:type`.
See https://en.wikipedia.org/wiki/RDF_Schema

**Comment by DhanshreeA** [comment]
@ashleysommer I believe @MuraliKrishna26 is talking about `RDF.Property` and is correct in pointing out that Property is in RDF Class and not RDFS. :) I am a bit rusty on the spec itself, but at least within this Python implementation `RDFS.Property` does not seem to be the case.

**Comment by ashleysommer** [comment]
> @MuraliKrishna26 is talking about RDF.Property and is correct in pointing out that Property is in RDF Class and not RDFS

Yes, you're right. I misread what @MuraliKrishna26 had written.
